diff --git i/top/maint.mk w/top/maint.mk
index 4cbd5f4..2228a37 100644
--- i/top/maint.mk
+++ w/top/maint.mk
@@ -279,7 +279,7 @@ define _sc_search_regexp
    if test -n "$$files"; then						\
      if test -n "$$prohibit"; then					\
        grep $$with_grep_options $(_ignore_case) -nE "$$prohibit" $$files \
-         | grep -vE "$${exclude-^$$}"					\
+         | grep -vE "$${exclude:-^$$}"					\
          && { msg="$$halt" $(_sc_say_and_exit) } || :;			\
      else								\
        grep $$with_grep_options $(_ignore_case) -LE "$$require" $$files \
@@ -455,7 +455,8 @@ sc_prohibit_quotearg_without_use:

 # Don't include quote.h unless you use one of its functions.
 sc_prohibit_quote_without_use:
-	@h='quote.h' re='\<quote(_n)? *\(' $(_sc_header_without_use)
+	@h='quote.h' re='\<quote((_n)? *\(|_quoting_options\>)' \
+	  $(_sc_header_without_use)

 # Don't include this header unless you use one of its functions.
 sc_prohibit_long_options_without_use:
@@ -1332,7 +1333,7 @@ alpha beta stable: $(local-check) writable-files $(submodule-checks)
 	$(MAKE) vc-diff-check
 	$(MAKE) news-check
 	$(MAKE) distcheck
-	$(MAKE) dist XZ_OPT=-9ev
+	$(MAKE) dist
 	$(MAKE) $(release-prep-hook) RELEASE_TYPE=$@
 	$(MAKE) -s emit_upload_commands RELEASE_TYPE=$@

